daemon/device_unix_test.go

```diff
@@ -0,0 +1,146 @@
+// +build !windows
+
+package daemon
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/containerd/containerd/pkg/userns"
+	"github.com/opencontainers/runc/libcontainer/devices"
+)
+
+func cleanupTest() {
+	ioutilReadDir = ioutil.ReadDir
+	usernsRunningInUserNS = userns.RunningInUserNS
+}
+
+// Based on test from runc (libcontainer/devices/device_unix_test.go).
```

Comment:
For these as well. 

For permalinks, It would be good to provide a permalink to the actual code / lines that these were based on, e.g. https://github.com/opencontainers/runc/blob/v1.0.0/libcontainer/devices/device_unix.go#L76-L120 for the earlier example.